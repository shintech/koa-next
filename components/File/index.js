import PropTypes from 'prop-types'
import Wrapper from './wrapper'

const File = ({ filename = '' }) =>
  <Wrapper>
    <a key={filename} href={`static/files/${filename}`}>{filename}</a>
  </Wrapper>

File.propTypes = {
  filename: PropTypes.string.isRequired
}

export default File
