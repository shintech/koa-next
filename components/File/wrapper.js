import styled from 'styled-components'

const Wrapper = styled.div`
  a {
    color: lightgreen;
    
    :visited {
      color: magenta;
    }
  }
`

export default Wrapper
