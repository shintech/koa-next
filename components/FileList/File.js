import React from 'react'
import SVG from 'static/images/react.svg'

const File = ({ file, selectNowPlaying }) => {
  return (
    <a href={`static/files/${file}`} onClick={e => {
      e.preventDefault()
      selectNowPlaying(file)
    }}>
      <div className='cover'><SVG /></div>
      <div className='info'>{file}</div>
    </a>
  )
}

export default File