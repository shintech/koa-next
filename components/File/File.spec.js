/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import File from 'components/File'

const shallowExpect = compose(expect, toJSON, shallow)

// describe('COMPONENT -> Content...', () => {
//   let clicker = shallow(<Content content='Hello World!' />)

//   it(`expect Content p text to equal "Hello World!"...`, () => {
//     expect(clicker.find('p').text()).toBe('Hello World!')
//   })
// })

describe('COMPONENT -> Content - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<File filename='file.mp3' />).toMatchSnapshot()
  })
})
