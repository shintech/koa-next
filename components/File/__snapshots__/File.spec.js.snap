// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Content - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper>
  <a
    href="static/files/file.mp3"
    key="file.mp3"
  >
    file.mp3
  </a>
</wrapper__Wrapper>
`;
